modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheConfigurationSplitterImpl.java

```diff
@@ -41,11 +44,32 @@
     /** Marshaller. */
     private final Marshaller marshaller;
 
+    /** Grid kernal context. */
+    private final GridKernalContext ctx;
+
     /**
      * @param marshaller Marshaller.
+     * @param ctx Grid kernal context.
      */
-    public CacheConfigurationSplitterImpl(Marshaller marshaller) {
+    public CacheConfigurationSplitterImpl(Marshaller marshaller, GridKernalContext ctx) {
         this.marshaller = marshaller;
```

Comment:
let's add assertions for arguments
assert marshaller != null;
assert ctx != null;

Reply:
I strongly disagree. I am against this unnecessary asserts, this leads to bloated bytecode, nothing more

Reply:
ok, it's just a suggestion.
For me It seems that the code with the assertions is more concise and readable and I don't see any problem with "bloated bytecode".

Reply:
+1 with Ivan here, if it`s necessary u may add @Nullable and check them, but not asserts.